[2024-08-04T10:17:36.940+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: excel_to_oracle_dag.load_excel_to_oracle manual__2024-08-04T10:17:35.508890+00:00 [queued]>
[2024-08-04T10:17:36.949+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: excel_to_oracle_dag.load_excel_to_oracle manual__2024-08-04T10:17:35.508890+00:00 [queued]>
[2024-08-04T10:17:36.949+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 2
[2024-08-04T10:17:36.964+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): load_excel_to_oracle> on 2024-08-04 10:17:35.508890+00:00
[2024-08-04T10:17:36.971+0000] {standard_task_runner.py:57} INFO - Started process 249 to run task
[2024-08-04T10:17:36.974+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'excel_to_oracle_dag', 'load_excel_to_oracle', 'manual__2024-08-04T10:17:35.508890+00:00', '--job-id', '24', '--raw', '--subdir', 'DAGS_FOLDER/excel_to_oracle_dag.py', '--cfg-path', '/tmp/tmpsj_xs5az']
[2024-08-04T10:17:36.975+0000] {standard_task_runner.py:85} INFO - Job 24: Subtask load_excel_to_oracle
[2024-08-04T10:17:37.021+0000] {task_command.py:416} INFO - Running <TaskInstance: excel_to_oracle_dag.load_excel_to_oracle manual__2024-08-04T10:17:35.508890+00:00 [running]> on host 290a486cedb7
[2024-08-04T10:17:37.093+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='excel_to_oracle_dag' AIRFLOW_CTX_TASK_ID='load_excel_to_oracle' AIRFLOW_CTX_EXECUTION_DATE='2024-08-04T10:17:35.508890+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-08-04T10:17:35.508890+00:00'
[2024-08-04T10:17:37.104+0000] {base.py:73} INFO - Using connection ID 'Connection_bai_test' for task execution.
[2024-08-04T10:17:37.179+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/compat/_optional.py", line 142, in import_optional_dependency
    module = importlib.import_module(name)
  File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'openpyxl'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/excel_to_oracle_dag.py", line 49, in load_excel_to_oracle
    excel_data = pd.ExcelFile(file_path)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/excel/_base.py", line 1513, in __init__
    self._reader = self._engines[engine](self._io, storage_options=storage_options)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/excel/_openpyxl.py", line 548, in __init__
    import_optional_dependency("openpyxl")
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/compat/_optional.py", line 145, in import_optional_dependency
    raise ImportError(msg)
ImportError: Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.
[2024-08-04T10:17:37.191+0000] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=excel_to_oracle_dag, task_id=load_excel_to_oracle, execution_date=20240804T101735, start_date=20240804T101736, end_date=20240804T101737
[2024-08-04T10:17:37.202+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 24 for task load_excel_to_oracle (Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.; 249)
[2024-08-04T10:17:37.226+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-08-04T10:17:37.243+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
